File: Java/src/ch08/RecurBinaryTree.java
package ch08;

public class RecurBinaryTree {

    public static class Node{
        public int value;
        public Node left;
        public Node right;

        public Node(int v){
            value = v;
        }
    }

    public static void recur(Node root){
        if (null == root){
            return;
        }
        System.out.print(root.value);
        recur(root.left);
        System.out.print(root.value);
        recur(root.right);
        System.out.print(root.value);

    }

    public static void main(String[] args) {
        Node root = new Node(6);
        root.left = new Node(4);
        root.right = new Node(8);
        root.left.left = new Node(2);
        root.left.right = new Node(5);
        root.right.left = new Node(7);
        root.right.right = new Node(9);
        root.left.left.left = new Node(1);
        root.left.left.right = new Node(3);

        recur(root);
    }

}

File: Python/ch08/RecurBinaryTree.py
# 二叉树的节点
class Node:
    def __init__(self, v):
        self.left = None
        self.right = None
        self.value = v


def recur(r):
    if r is None:
        return
    print(r.value, end='')
    recur(r.left)
    print(r.value, end='')
    recur(r.right)
    print(r.value, end='')


if __name__ == '__main__':
    root = Node(6)
    root.left = Node(4)
    root.right = Node(8)
    root.left.left = Node(2)
    root.left.right = Node(5)
    root.right.left = Node(7)
    root.right.right = Node(9)
    root.left.left.left = Node(1)
    root.left.left.right = Node(3)

    recur(root)
